src/main/java/co/rsk/federate/watcher/FederationWatcherListener.java

```diff
@@ -0,0 +1,31 @@
+package co.rsk.federate.watcher;
+
+import co.rsk.peg.federation.Federation;
+import java.util.Optional;
+
+/**
+ * A listener interface for receiving notifications about changes in federations.
+ * Implementers of this interface will be notified when the active or retiring federation changes.
+ */
+public interface FederationWatcherListener {
+
+    /**
+     * Invoked when the active federation changes.
+     *
+     * @param oldFederation an {@code Optional} containing the previous active federation.
+     *                      This will be empty if there was no active federation before.
+     * @param newFederation an {@code Optional} containing the new active federation.
+     *                      This will be empty if there is no active federation after the change.
+     */
+    void onActiveFederationChange(Optional<Federation> oldFederation, Optional<Federation> newFederation);
```

Comment:
This is true, I have changed the interface design since if we know there is a change then we also know the value to be guranteed. So it does not make sense to have optionals as params. Let me know what you think.